packages/adapter-firebase/src/index.ts

```diff
@@ -1,283 +1,250 @@
-import type firebase from "firebase"
-import { createHash, randomBytes } from "crypto"
-import { Adapter } from "next-auth/adapters"
+import { FirebaseOptions, initializeApp } from 'firebase/app';
 import {
-  querySnapshotToObject,
-  docSnapshotToObject,
-  stripUndefined,
-} from "./utils"
-import { Profile, Session, User } from "next-auth"
-
-interface FirebaseVerificationRequest {
-  id: string
-  identifier: string
-  token: string
-  expires: Date
+  addDoc,
+  collection,
+  deleteDoc,
+  doc,
+  getDoc,
+  getDocs,
+  getFirestore,
+  limit,
+  query,
+  runTransaction,
+  setDoc,
+  where,
+  connectFirestoreEmulator
+} from 'firebase/firestore';
+import { Account } from 'next-auth';
+import { Adapter, AdapterSession, AdapterUser, VerificationToken } from 'next-auth/adapters';
+
+import { getConverter } from './converter';
+
+type IndexableObject = Record<string, unknown>;
+
+interface FirestoreAdapterOptions {
+  emulator?: {
+    host?: string;
+    port?: number;
+  };
 }
 
-export type FirebaseSession = Session & {
-  id: string
-  expires: Date
-}
+/**
+ * @deprecated Use FirestoreAdapter export instead
+ */
+export const FirebaseAdapter = ({ emulator, ...firebaseOptions }: FirebaseOptions & FirestoreAdapterOptions): Adapter => {
```

Comment:
Since we are doing a breaking change, let's just drop it.

```suggestion
/**
 * @deprecated Use FirestoreAdapter export instead
 */
export function FirestoreAdapter({ emulator, ...firebaseOptions }: FirebaseOptions & FirestoreAdapterOptions): Adapter {
```